### PR TITLE
simplify location logic, let Database take care of allocation

### PR DESCRIPTION
**Pull request by ralphtheninja**
See comments below.

### CODE REVIEW

**Comment by juliangruber** [comment]
you got rid of 12 sloc, like a real c pro :)

**Comment by ralphtheninja** [comment]
Ack I rebased it and lost all the comments. Gah.

File: src/database.h [hunk 1]
```diff
@@ -73,19 +73,18 @@ class Database : public node::ObjectWrap {
   const leveldb::Snapshot* NewSnapshot ();
   void ReleaseSnapshot (const leveldb::Snapshot* snapshot);
   void CloseDatabase ();
-  NanUtf8String* Location();
   void ReleaseIterator (uint32_t id);
 
-  Database (NanUtf8String* location);
+  Database (v8::Handle<v8::Value> from);
```
Comment (rvagg) [inline src/database.h]: should it be `&from`?

Comment (ralphtheninja) [inline src/database.h]: Aye we could do that, or `const &`

Comment (ralphtheninja) [inline src/database.h]: Updated.

**Comment by ralphtheninja** [comment]
Anyone against? I'd like to merge this today.

**Pull request merged**